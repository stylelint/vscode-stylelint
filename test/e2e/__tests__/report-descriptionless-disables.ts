import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('"stylelint.reportDescriptionlessDisables" setting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should report invalid-scope disables when enabled', async () => {
		const { document } = await openDocument('descriptionless-disables/test.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: '--report-descriptionless-disables',
				message: 'Disable for "color-hex-length" is missing a description',
				range: [2, 4, 2, 53],
				severity: 'error',
			},
		]);
	});
});
