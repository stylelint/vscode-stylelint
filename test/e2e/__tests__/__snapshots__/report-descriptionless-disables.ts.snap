// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.reportDescriptionlessDisables" setting should report invalid-scope disables when enabled 1`] = `
[
  {
    "code": "--report-descriptionless-disables",
    "message": "Disable for "function-name-case" is missing a description",
    "range": {
      "end": {
        "character": 55,
        "line": 2,
      },
      "start": {
        "character": 4,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
